### ISSUE TITLE
How to use the function named classify_tweet in chapter5?

### ISSUE THREAD

**Issue by laohuijiadezhu** [falloutdurham/beginners-pytorch-deep-learning#55]
The variable model in the function don't be parament. How to use?
![image](https://user-images.githubusercontent.com/59173157/105319833-62117a00-5c00-11eb-8fc2-960ccd47ad6e.png)

**Comment by MarcusFra** [comment]
Hi, you can use the function like
```
your_tweet = "I really liked it!"
classify_tweet(your_tweet)
```
But I think you are right. Since `model` is set to eval() mode, it should be included as an argument, as I suppose it's not affected from the model.eval() call inside the function.
Could you please check whether `model.training` is True or False after running the classify_tweet() function? (it still might be set to eval mode from the training loop, where it's validated on the validation set)

**Comment by franzko03** [comment]
Hi, you can test with:
for i in range(1,10):    
    myTweet= tweetsDF.values[i,5]
    kategorie= classify_tweet(myTweet)
    print(kategorie + ':' + myTweet)

tweetsDF is  a  pandas.DataFrame

**Comment by franzko03** [comment]
hi,
with torchtext 0.8.1 chapter 5 skript works as follows:

(sample size and vocab size is arbitrary)
`tweetsDF.sample(20000).to_csv("train-processed-sample.csv", header=None, index=None)

LABEL = data.LabelField()
TWEET = **data.Field(tokenizer_language='en', lower=True)**

fields = [('score',None), ('id',None),('date',None),('query',None),
      ('name',None),
      ('tweet', TWEET),('category',None),('label',LABEL)]

and :
print((len(train),len(test),len(valid)))
print(vars(train.examples[7]))
vocab_size = 40000
TWEET.build_vocab(train, max_size = vocab_size)
LABEL.build_vocab(train)
print(len(TWEET.vocab))
print(TWEET.vocab.freqs.most_common(10))`

with the results
(12000, 4000, 4000)
{'tweet': ['@nicnicole', 'nah', 'i', 'followed', 'them', 'first', 'then', 'they', 'followed', 'me.', 'i', 'think', '12-306', 'overnight', 'is', 'pretty', 'good'], 'label': '1'}
30846
[('i', 5658), ('to', 4204), ('the', 3759), ('a', 2714), ('my', 2352), ('and', 2190), ('you', 1807), ('is', 1654), ('in', 1533), ('for', 1524)]

**Comment by MarcusFra** [comment]
Thanks for mentioning and your suggestion, @franzko03! 

True, the definition as `TWEET = data.Field(tokenize='spacy', lower=True)` raises an exception after upgrading to torchtext 0.8.1. But I think here are two things at work:

1.) I think the change you've proposed (tokenizer_language='en') is likely due to an upgrade/install of spaCy v3.0. Since that release language abbreviations as 'en' or 'de' are deprecated and have to be replaced by the model name, e. g. 'en_core_web_sm'.

2.) Since we are using the `Field class` (which in turn is currently being deprecated), the tokenizer_language attribute's default value is 'en', which leads to an error, as mentioned, due to a new definition in spacy v3. For being able to use the spacy tokenizer with the Field class use

```
TWEET = data.Field('spacy', tokenizer_language='en_core_web_sm', lower=True)
```

 or

```
from torchtext.data.utils import get_tokenizer

tokenizer = get_tokenizer('spacy', language='en_core_web_sm')
TWEET = data.Field(tokenizer, lower=True)
```

The use of `tokenizer_language='en'` likely leads to a different tokenization, as torchtext's `data.utils._basic_english_normalize` function is being called.


Aside that, the deprecated parts in the example (Field class etc.) need to be updated soon as they are now moved to the `torchtext.legacy` folder. I am going to open a separate issue on that the coming weekend.
